Next Version of sBTC Clarity Scaffolding

setzeus opened the pull request:
# Scaffolding For Next Version of sBTC Clarity Contracts
## Summary of Changes

This PR is tied to #353 & is an attempt to scaffold the next version of sBTC Clarity contracts based on the design researched & reviewed by the community. There is no code yet in any of the contracts. 

## Testing
No testing as none of the contracts included here have anything to test.

### Risks
This appears relatively risk-free since the only possible this can effect Romeo is if a directory change broke items needed along some process.

### How were these changes tested?
They weren't since there is no code inside any of these contracts yet.

## Checklist:

- [x ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

moodmosaic commented on clarity/next/package.json:
```diff
@@ -0,0 +1,22 @@
+    "@stacks/transactions": "^6.9.0",
+    "chokidar-cli": "^3.0.0",
+    "typescript": "^5.2.2",
+    "vite": "^4.4.9",
```
fast-check is missing, should be added.

setzeus replied:
What version?

moodmosaic replied:
In this repo, for romeo, I've been using `"fast-check": "^3.13.1"` with no issues.

setzeus replied:
Included!

moodmosaic commented on clarity/next/tests/sbtc-db_test.ts:
```diff
@@ -0,0 +1,26 @@
+
+import { Clarinet, Tx, Chain, Account, types } from 'https://deno.land/x/clarinet@v1.7.1/index.ts';
```
Should we have deno/clarinet v1 tests if we’re going g to use the new clarinet-sdk?

friedger replied:
No, should be sdk only

moodmosaic replied:
Then those tests should be removed, I guess.

setzeus replied:
Copy, so remove all .ts files in the /tests directory?

moodmosaic replied:
All `*_test.ts` files, from what I see. The `contract.test.ts` uses the new clarity-sdk so it's fine.

setzeus replied:
Removed

friedger commented:
Why `next` folder? Will there be a `main` folder as well?

setzeus commented:
@friedger negative, changing to sbtc-nakamoto. Made that name as a place holder.

setzeus committed:
removed deno tests, renamed directory & added fast-check

moodmosaic committed:
next version of sbtc scaffolding

# Conflicts:
#	clarity/romeo/asset-contract/tests/asset_GetBitcoinWalletPublicKeyCommand.ts
#	clarity/romeo/asset-contract/tests/asset_SetBitcoinWalletPublicKeyCommand.ts
#	clarity/romeo/asset-contract/tests/asset_SetBitcoinWalletPublicKeyCommand_NonOwner.ts

moodmosaic reviewed (dismissed):
<del>LGTM (for the changes in romeo/asset-contract)</del> https://github.com/stacks-network/sbtc/pull/354#event-10969899345

friedger reviewed (commented):
Why is romeo rust binary under `clarity`?
Top-level clarity folder does not make sense to me :-)

friedger commented on clarity/nakamoto/settings/Devnet.toml:
```diff
@@ -0,0 +1,150 @@
+# epoch_2_4 = 105
+
+
+# Send some stacking orders
```
Why do we need stacking order here?

friedger commented on clarity/nakamoto/package.json:
Clarinet.toml is missing

friedger commented on clarity/nakamoto/package.json:
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "next-tests",
```
Better name here

friedger reviewed (commented):
What is the acceptance criteria here?
Should we at least be able to run `clarinet check` and `yarn test`

setzeus commented:
> Why is romeo rust binary under `clarity`? Top-level clarity folder does not make sense to me :-)

The idea here is to create a single Clarity directory that holds different versions of sBTC - both the previous (Romeo) & new (Nakamoto) versions in this case.

Re: Rust binary was already in /src file in when copied over but you're right, we might as well remove /src & other files in that level since all Clarity items are in the "asset-contract" directory.

moodmosaic commented:
> What is the acceptance criteria here? Should we at least be able to run `clarinet check` and `yarn test`

@friedger I checked both when I rebased (and force-pushed) for `romeo/asset-contract` using Clarinet 1.8.0:

```
$ clarinet check
✔ 5 contracts checked

$ npm install && npm test

 ✓ tests/asset.test.ts (6) 26195ms
 ✓ tests/clarity-bitcoin-mini-deploy.test.ts (1)
```

And we still need to port the stuff in `ext` and `scripts` for the tests written in Clarity so they can run with the new SDK.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/stacks-network/sbtc-developer-release?pullRequest=354) <br/>All committers have signed the CLA.
